src/app/gagestats/gagepage/gagepage.component.html

```diff
@@ -120,106 +122,116 @@
+							<td>
+								<span *ngIf="!c.isEditing">{{c.variableType.name}}</span>
+								<span *ngIf="c.isEditing">
+									<ng-select name="variableTypeID" [(ngModel)]="c.variableTypeID" [clearable]="false">
 										<ng-option *ngFor="let variable of variables" [value]="variable.id">{{variable.name}}</ng-option> 
 									</ng-select>
 								</span>
 							</td>
 							<td>
-								<span *ngIf="!newChar.isEditing">{{newChar.value}}</span>
-								<span *ngIf="newChar.isEditing"><input type="number" [(ngModel)]="newChar.value" size="2" class="input-100"></span>
+								<span *ngIf="!c.isEditing">{{c.value}}</span>
+								<span *ngIf="c.isEditing"><input type="number" [(ngModel)]="c.value" size="1" class="input-100"></span>
 							</td>
 							
-							<td class="160width">
-								<span *ngIf="!newChar.isEditing">{{newChar.unitType.name}}</span>
-								<span *ngIf="newChar.isEditing">
-									<ng-select name="unitTypeID" [(ngModel)]="newChar.unitTypeID" [clearable]="false">
+							<td>
+								<span *ngIf="!c.isEditing">{{c.unitType.name}}</span>
+								<span *ngIf="c.isEditing">
+									<ng-select name="unitTypeID" [(ngModel)]="c.unitTypeID" [clearable]="false">
 										<ng-option *ngFor="let unit of units" [value]="unit.id">{{unit.name}}</ng-option>
 									</ng-select>
 								</span>
 							</td>
 							<td>
-								<span *ngIf="!newChar.isEditing">{{newChar.citationID}}</span>
-								<span *ngIf="newChar.isEditing"><input type="number" [(ngModel)]="newChar.citationID" size="2"></span>
+								<span *ngIf="!c.isEditing">{{c.citationID}}</span>
+								<span *ngIf="c.isEditing"><input type="number" [(ngModel)]="c.citationID" size="2" class="input-100"></span>
                             </td>
                             <td>
-								<span *ngIf="!newChar.isEditing">{{newChar.comments}}</span>
-								<span *ngIf="newChar.isEditing"><textarea [(ngModel)]="newChar.comments" size="2"></textarea></span>
+								<span *ngIf="!c.isEditing">{{c.comments}}</span>
+								<span *ngIf="c.isEditing"><textarea [(ngModel)]="c.comments" size="2"></textarea></span>
 							</td>
 						</tr>
-						</tbody>
+					</tbody>
 				</table>
 			</div>
 		</div>
 		<!--Streamflow Statistics Table-->
 		<div>
 			<div class="table-title">Streamflow Statistics
-				<button *ngIf="loggedInRole && editGage" type="button" class="button small blue mright-xs"  (click)="addStreamflowStatistic()">Add Streamflow Statistic</button>
+				<div class="stat-filters">
+					<div><label>Filter by Statistic Group: </label></div>
+					<div>
+						<ss-multiselect-dropdown [options]="filteredStatGroups" [texts]="myMSTexts" [settings]="myRTSettings"
+						[(ngModel)]="selectedStatGroup">
+						</ss-multiselect-dropdown>
+					</div>
+					<div><button class="button small grey" *ngIf="!preferred" (click)="setPreferred()">Show Preferred Stats  <i class="fa fa-toggle-off" ></i></button></div>
```

Comment:
Instead of having a `setPreferred()` function, you can set the preferred item in the html using `(click)="preferred = true"` or `(click)="preferred = false"`

Reply:
Good suggestion. I am learn more about functions that can be preferred in the html files compared to the typescript components.